spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/SpannerKeyWriter.java

```diff
@@ -0,0 +1,44 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.core.convert;
+
+import com.google.cloud.spanner.Key;
+
+/**
+ * An interface that can create Cloud Spanner keys and utlizes conversion.
+ *
+ * @author Chengyuan Zhao
+ *
+ * @since 1.1
+ */
+public interface SpannerKeyWriter {
```

Comment:
I'm not sure why this interface is useful vs. keeping these methods in `SpannerEntityWriter`.

Reply:
The reason is this new one is what's added to the constructor of SpannerPersistentEntityImpl because it is needed to convert  the properties that comprise the composite key.

Reply:
Why would giving `SpannerEntityWriter` to the `SpannerPersistentEntityImpl` be so bad? You never implement  just `SpannerKeyWriter`, and it looks like an internal class.

Reply:
To be honest I personally don't think it's bad, but I did it because I thought if I gave it a whole entity-converter interface to something that only uses the key-converter functions would offend somebody with more tender java sensibilities. I can get rid of the interface.

Reply:
done